VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -62,15 +65,26 @@ void ViewItem::insertColumn(int column)
 
 void ViewItem::insertNode(Model::Node* node, int column, int row)
 {
+	auto ref = new Model::NodeReference();
+	ref->setTarget(node);
```

Comment:
When you make the new node type, just make it have a constructor that already takes the reference, instead of calling setTarget separately.